sled-agent/src/instance.rs

```diff
@@ -2110,20 +2122,16 @@ mod tests {
             .await
             .expect("failed to send Instance::put_state");
 
-        // Timeout our future waiting for the instance-state-change at
-        // `TIMEOUT_DURATION`, which should fail because zone boot will take
-        // twice that by construction.
-        let timeout_fut = timeout(TIMEOUT_DURATION, put_rx);
-
-        // And advance time by twice that, so that the actual
-        // `MockZones::boot()` call should be exercised (or will be soon).
-        tokio::time::advance(TIMEOUT_DURATION * 2).await;
-
-        tokio::time::resume();
-
+        // Timeout our future waiting for the instance-state-change at 1s. This
+        // is much shorter than the actual `TIMEOUT_DURATION`, but the test
+        // structure requires that we actually wait this period, since we cannot
+        // advance time manually in a multi-threaded runtime.
+        let timeout_fut = timeout(TIMEOUT, put_rx);
+        println!("Awaiting zone-boot timeout");
```

Comment:
(should these println's be going into logctx instead?)

Reply:
They can, yeah. I wanted to see them during runs with `--nocapture`, but I can move them over if that's preferred.

Reply:
Moved in a0b73002c